v2/spiffetls/tlsconfig/config_test.go

```diff
@@ -0,0 +1,687 @@
+	}{
+		{
+			name: "success",
+			source: &fakeSource{
+				err: nil,
+				svid: &x509svid.SVID{
+					ID: spiffeid.Must("test.domain", "host"),
+					Certificates: []*x509.Certificate{
+						{Raw: []byte("body")},
+					},
+				},
+			},
+			expectedCerts: [][]byte{[]byte("body")},
+		},
+		{
+			name: "source return error",
+			source: &fakeSource{
+				err: errors.New("some error"),
+			},
+			err: "some error",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			getCertificate := tlsconfig.GetCertificate(testCase.source)
+			require.NotNil(t, getCertificate)
+
+			//  Execute getCertificate
+			tlsCert, err := getCertificate(&tls.ClientHelloInfo{})
+			if testCase.err != "" {
+				require.EqualError(t, err, testCase.err)
+				require.Nil(t, tlsCert)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, testCase.expectedCerts, tlsCert.Certificate)
+		})
+	}
+}
+
+func createTestTlsConfig() *tls.Config {
+	tlsCert := tls.Certificate{Certificate: [][]byte{[]byte("body")}}
+	return &tls.Config{
+		Certificates: []tls.Certificate{
+			tlsCert,
+		},
+		NameToCertificate: map[string]*tls.Certificate{"cert": &tlsCert},
+		GetCertificate: func(info *tls.ClientHelloInfo) (certificate *tls.Certificate, err error) {
+			return nil, nil
+		},
+		GetClientCertificate: func(info *tls.CertificateRequestInfo) (certificate *tls.Certificate, err error) {
+			return nil, nil
+		},
+		VerifyPeerCertificate: nil,
+		RootCAs:               x509.NewCertPool(),
+		ClientAuth:            tls.RequestClientCert,
+	}
+}
+
+func TestGetClientCertificate(t *testing.T) {
+	testCases := []struct {
+		name          string
+		source        *fakeSource
+		err           string
+		expectedCerts [][]byte
+	}{
+		{
+			name: "success",
+			source: &fakeSource{
+				err: nil,
+				svid: &x509svid.SVID{
+					ID: spiffeid.Must("test.domain", "host"),
+					Certificates: []*x509.Certificate{
+						{Raw: []byte("body")},
+					},
+				},
+			},
+			expectedCerts: [][]byte{[]byte("body")},
+		},
+		{
+			name: "source return error",
+			source: &fakeSource{
+				err: errors.New("some error"),
+			},
+			err: "some error",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			getClientCertificate := tlsconfig.GetClientCertificate(testCase.source)
+			require.NotNil(t, getClientCertificate)
+
+			//  Execute getCertificate
+			tlsCert, err := getClientCertificate(&tls.CertificateRequestInfo{})
+			if testCase.err != "" {
+				require.EqualError(t, err, testCase.err)
+				require.Nil(t, tlsCert)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, testCase.expectedCerts, tlsCert.Certificate)
+		})
+	}
+}
+
+func TestVerifyPeerCertificate(t *testing.T) {
+	td := spiffeid.RequireTrustDomainFromString("domain1.test")
+	ca1 := test.NewCA(t)
+	bundle1 := ca1.Bundle(td)
+	require.NotNil(t, bundle1)
```

Comment:
nit: this assertion is overkill... the test CA already does assertions internally

Reply:
done